include/RAJA/pattern/params/reducer.hpp

```diff
@@ -101,48 +95,99 @@ namespace detail
     unsigned int * device_count = nullptr;
 #endif
 
-    using ARG_TUP_T = camp::tuple<value_type*>;
-    RAJA_HOST_DEVICE ARG_TUP_T get_lambda_arg_tup() { return camp::make_tuple(&val); }
+    // These are types and parameters extracted from this struct, and given to the forall.
+    using ARG_TUP_T = camp::tuple<VType*>;
+    RAJA_HOST_DEVICE ARG_TUP_T get_lambda_arg_tup() { return camp::make_tuple(&valop_m); }
 
     using ARG_LIST_T = typename ARG_TUP_T::TList;
     static constexpr size_t num_lambda_args = camp::tuple_size<ARG_TUP_T>::value ;
   };
 
-} // namespace detail
+  // Partial specialization of Reducer for ValLoc
+  // T must be a ValLoc
+  // TTOp is the template template version of Op, required by ValOp
+  // VType must be ValOp<ValLoc<>,Op>
+  template <typename Op, typename T, template <typename, typename, typename> class TTOp>
+  struct Reducer<Op, T, ValOp<ValLoc<typename T::value_type, typename T::index_type>, TTOp>> : public ForallParamBase {
```

Comment:
It might be easier to do this here.
```suggestion
  template <typename T, typename Index, template <typename, typename, typename> class Op>
  struct Reducer<Op<ValLoc<T, Index>, ValLoc<T, Index>, ValLoc<T, Index>>, ValLoc<T, Index>, ValOp<ValLoc<T, Index>, Op>> : public ForallParamBase {
```